pkg/apis/serving/v1/configuration_defaults.go

```diff
@@ -23,20 +23,49 @@ import (
 	"knative.dev/serving/pkg/apis/serving"
 )
 
+type configSpecKey struct{}
+
+// WithConfigurationSpec stores a ConfigurationSpec in the context, to allow
+// ConfigurationSpec.SetDefaults to determine whether the update would create a
+// new Revision.
+func WithConfigurationSpec(ctx context.Context, spec *ConfigurationSpec) context.Context {
```

Comment:
nit: I'm not a fan of the name I'm suggesting - but it's a bit clear that previous config spec is being stored/lookedup from the context.

```suggestion
func WithPreviousConfigurationSpec(ctx context.Context, spec *ConfigurationSpec) context.Context {
```